src/App.js

```diff
@@ -0,0 +1,38 @@
+import { Header } from "./components/Header";
+import MainContent from "./components/MainContent";
+import Footer from "./components/Footer";
+import Login from "./components/Login";
+import Signup from "./components/Signup";
+import { SubmitIdea } from "./components/SubmitIdea";
+import { Selection } from "./components/Selection";
+import "bootstrap/dist/css/bootstrap.min.css";
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
+import Parse from "parse";
+
+function App() {
+  return (
+    <div className="App">
+      <Header />
+      <BrowserRouter>
+        <Routes>
+          <Route path="/signup" element={!Parse.User.current( ) ? <Signup /> : <Navigate to='/'/>} />
+          <Route path="/login" element={!Parse.User.current( ) ? <Login /> : <Navigate to='/'/>} />
+          <Route path="/submitIdea" element={Parse.User.current( ) ? <SubmitIdea /> : <Navigate to='/login'/>} />
+          <Route path="/selection" element={Parse.User.current( ) ? <Selection /> : <Navigate to='/login'/>} />
```

Comment:
That's one way to do it.
Try to see if you can limit your identical occurrences of conditional logic.
If you have the same conditions, can you somehow group them?